### ISSUE TITLE
Test code is in the ansible production code breaking Windows installation among other things

### ISSUE THREAD

**Issue by kwilson7770** [ansible/ansible#34827]
This is a installation path for ansible:

test/integration/targets/copy/files/subdir/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/circles/subdir1/bar.txt

Based on the name, this was a test that was put into the main code and is causing this to break Windows installations

I already commented on issue https://github.com/ansible/ansible/issues/31419 in regards to this, but the issue was closed saying there is no support for Windows instead of looking into these excessively long directories

I found a reference to the possible source of this code here:

https://github.com/ansible/ansible/blob/1b0c44584624db74921e053c85333a6c6e54501c/test/integration/targets/copy/tasks/tests.yml

**Comment by kwilson7770** [comment]
Also, you can find where this code was added on fossies.org:

https://fossies.org/diffs/ansible/2.3.2.0_vs_2.4.0.0/

look for 

> test/integration/targets/copy

 and you will see where it was added

**Comment by ansibot** [comment]
@kwilson7770 Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- issue type
- ansible version
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by alikins** [comment]
> Based on the name, this was a test that was put into the main code and is causing this to break Windows installations

What happens on install?  Are there error messages? If so, what are they?

What installation method is being used? 

Which ansible version?

**Comment by jhawkesworth** [comment]
@kwilson7770 What actual problem is being caused here?  Since you can't use ansible installed on windows in this way and there's no support for running ansible directly from a windows machine then its not clear what is being attempted.

If your goal is to work on the ansible code so that it can run on windows, then I suggest installing the source from git.  Yes, the packaging issue remains but there is little point in fixing it until much larger issues which stop windows from being used as the controller are solved.

If you want to run ansible from a windows machine, while still unsupported, I can vouch for it working well if you use Windows Subsystem for Linux by installing 'bash on windows' (available in recent Windows 10 versions).

**Comment by jborean93** [comment]
@kwilson7770 ultimately this issue is due to the copy tasks testing out recursive symbolic links that was added for 2.4. While we do not support Ansible running on Windows and these are some of the unfortunately side effects of our testing, I did create a PR to fix this for future releases https://github.com/ansible/ansible/pull/35073.

Instead of storing the symlink in the repo, it will be created on the fly for the test itself.
